orchagent/dash/dashaclorch.cpp

```diff
@@ -0,0 +1,873 @@
+
+        if (task_status == task_need_retry)
+        {
+            SWSS_LOG_DEBUG(
+                "Task %s - %s need retry",
+                table_name.c_str(),
+                op.c_str());
+            ++itr;
+        }
+        else
+        {
+            if (task_status != task_success)
+            {
+                SWSS_LOG_WARN("Task %s - %s fail",
+                              table_name.c_str(),
+                              op.c_str());
+            }
+            else
+            {
+                SWSS_LOG_DEBUG(
+                    "Task %s - %s success",
+                    table_name.c_str(),
+                    op.c_str());
+            }
+
+            itr = consumer.m_toSync.erase(itr);
+        }
+    }
+
+    m_dash_acl_group_bulker.flush();
+    m_dash_acl_rule_bulker.flush();
+
+    callDeferFunctions();
+}
+
+task_process_status DashAclOrch::taskUpdateDashAclIn(
+    const string &key,
+    const TaskArgs &data)
+{
+    SWSS_LOG_ENTER();
+
+    return bindAclToEni(m_dash_acl_in_table, key, data);
+}
+
+task_process_status DashAclOrch::taskRemoveDashAclIn(
+    const string &key,
+    const TaskArgs &data)
+{
+    SWSS_LOG_ENTER();
+
+    return unbindAclFromEni(m_dash_acl_in_table, key, data);
+}
+
+task_process_status DashAclOrch::taskUpdateDashAclOut(
+    const string &key,
+    const TaskArgs &data)
+{
+    SWSS_LOG_ENTER();
+
+    return bindAclToEni(m_dash_acl_out_table, key, data);
+}
+
+task_process_status DashAclOrch::taskRemoveDashAclOut(
+    const string &key,
+    const TaskArgs &data)
+{
+    SWSS_LOG_ENTER();
+
+    return unbindAclFromEni(m_dash_acl_out_table, key, data);
+}
+
+task_process_status DashAclOrch::taskUpdateDashAclGroup(
+    const string &key,
+    const TaskArgs &data)
+{
+    SWSS_LOG_ENTER();
+
+    auto &acl_group = m_dash_acl_group_table[key];
+    bool is_existing = acl_group.m_dash_acl_group_id != SAI_NULL_OBJECT_ID;
+
+    vector<sai_attribute_t> attrs;
+
+    if (updateValue(data, "ip_version", acl_group.m_ip_version) || (acl_group.m_ip_version && !is_existing))
+    {
+        attrs.emplace_back();
+        attrs.back().id = SAI_DASH_ACL_GROUP_ATTR_IP_ADDR_FAMILY;
+        attrs.back().value.s32 = *(acl_group.m_ip_version);
+    }
+
+    // Guid wasn't mapping to any SAI attributes
+    updateValue(data, "guid", acl_group.m_guid);
+
+    if (!is_existing && attrs.empty())
+    {
+        // Assign default values
+        if (!acl_group.m_ip_version)
+        {
+            acl_group.m_ip_version = SAI_IP_ADDR_FAMILY_IPV4;
+            attrs.emplace_back();
+            attrs.back().id = SAI_DASH_ACL_GROUP_ATTR_IP_ADDR_FAMILY;
+            attrs.back().value.s32 = *(acl_group.m_ip_version);
+        }
+    }
+
+    if (!attrs.empty())
+    {
+        if (!is_existing)
+        {
+            // Create a new ACL group
+            acl_group.m_status = m_dash_acl_group_bulker.create_entry(&acl_group.m_dash_acl_group_id, static_cast<uint32_t>(attrs.size()), attrs.data());
```

Comment:
Typically when we use bulk operations there is an associated 'post' method that is called after the bulk operation is finished, e.g. `addRoute` and `addRoutePost` in route orch. Do we need the same pattern here?

Reply:
Done, I changed the bulk operation to normal SAI operation.